src/core/Grappa.sol

```diff
@@ -413,62 +460,236 @@ contract Grappa is OwnableUpgradeable, ReentrancyGuardUpgradeable, UUPSUpgradeab
+     * @notice settles cash token
+     * @param _account  who to settle for
+     * @param _tokenId  tokenId of option token to burn
+     * @param _amount   amount to settle
+     * @return payout what is being received
+     */
+    function _settleCashToken(address _account, uint256 _tokenId, uint64 _amount) internal returns (uint256 payout) {
+        address engine;
+        address collateral;
+
+        (engine, collateral, payout) = _getCashSettlement(_tokenId, _amount);
+
+        emit OptionSettled(_account, _tokenId, _amount, 0, payout);
+
+        // option owner gets collateral
+        if (payout != 0) ICashSettlement(engine).sendPayoutValue(collateral, _account, payout);
+    }
+
+    /**
+     * @notice settles physical token
      *
      * @param _account  who to settle for
      * @param _tokenId  tokenId of option token to burn
      * @param _amount   amount to settle
-     * @param _dryRun   bool to simulate transaction
-     * @return settlement struct
+     * @return debt what is owed
+     * @return payout what is being received
      */
-    function _settle(address _account, uint256 _tokenId, uint256 _amount, bool _dryRun)
+    function _settlePhysicalToken(address _account, uint256 _tokenId, uint64 _amount)
         internal
-        returns (Settlement memory settlement)
+        returns (uint256 debt, uint256 payout)
     {
-        uint64 amount = _amount.toUint64();
-
-        settlement = getSettlement(_tokenId, amount);
-
-        if (_dryRun) return settlement;
+        Settlement memory settlement = _getPhysicalSettlement(_tokenId, _amount);
 
         emit OptionSettled(_account, _tokenId, _amount, settlement.debt, settlement.payout);
 
         if (settlement.debt != 0 && settlement.payout != 0) {
-            settlement.tokenId = _tokenId;
-            settlement.tokenAmount = amount;
-            settlement.debtor = msg.sender;
-            settlement.creditor = _account;
-
-            IPhysicalSettlement(settlement.engine).settlePhysicalToken(settlement);
-        } else if (settlement.payout != 0) {
-            address payoutAsset = assets[settlement.payoutAssetId].addr;
-            IMarginEngine(settlement.engine).sendPayoutValue(payoutAsset, _account, settlement.payout);
+            IPhysicalSettlement engine = IPhysicalSettlement(settlement.engine);
+
+            (,,, uint64 expiry,,) = _tokenId.parseTokenId();
+
+            // if settlement window closed do nothing
+            if (block.timestamp > expiry + engine.getSettlementWindow()) return (0, 0);
+
+            TokenTracker storage tracker = tokenTracker[_tokenId];
+
+            // incrementing number of exercised token
+            tracker.exercised += _amount;
+
+            if (tracker.exercised > tracker.issued) revert GP_ExceedsIssuedTokens();
+
+            // issuer of option gets underlying asset (PUT) or strike asset (CALL)
+            engine.receiveDebtValue(assets[settlement.debtId].addr, msg.sender, settlement.debt);
```

Comment:
1. the `msg.sender` is design to be able to be callable by anyone here. so this is not the issuer.
2. you might also want to restrict this for physical settlement

Reply:
?

msg.sender is who is paying the debt. The debt is being held by the engine until the issuer (the account thats short) settles and receives the debt.